src/components/views/elements/RoleButton.js

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import AccessibleButton from './AccessibleButton';
+import dis from '../../../dispatcher';
+import sdk from '../../../index';
+
+export default React.createClass({
+    displayName: 'RoleButton',
+
+    propTypes: {
+        role: PropTypes.string.isRequired,
+        size: PropTypes.string,
+    },
+
+    getDefaultProps: function() {
+        return {
+            size: 25,
+            tooltip: false,
+        };
+    },
+
+    getInitialState: function() {
+        return {
+            showTooltip: false,
+        };
+    },
+
+    _onClick: function(ev) {
+        ev.stopPropagation();
+
+        let action;
+        switch(this.props.role) {
+            case 'start_chat':
+                action = 'view_create_chat';
+                break;
+            case 'room_directory':
+                action = 'view_room_directory';
+                break;
+            case 'create_room':
+                action = 'view_create_room';
+                break;
+            case 'home_page':
+                action = 'view_home_page';
+                break;
+            case 'settings':
+                action = 'view_user_settings';
+                break;
+        }
+        if (action) dis.dispatch({action: action});
+    },
+
+    _onMouseEnter: function() {
+        if (this.props.tooltip) this.setState({showTooltip: true});
+    },
+
+    _onMouseLeave: function() {
+        this.setState({showTooltip: false});
+    },
+
+    _getLabel() {
+        switch(this.props.role) {
```

Comment:
I wanted to avoid passing all 3 things through when they will only ever be the same values for a given type of button (ie. so you can't c+p fail and end up with a settings button that brings up the room directory)